test/apps/iodemo/io_demo.cc

```diff
@@ -567,32 +589,74 @@ static int set_data_size(char *str, options_t* test_opts)
     return 0;
 }
 
-static int parse_args(int argc, char **argv, options_t* test_opts)
+static int set_time(char *str, double *dest_p)
+{
+    char units[3] = "";
+    int num_fields;
+    double value;
+    double per_sec;
+
+    if (!strcmp(str, "inf")) {
+        *dest_p = std::numeric_limits<double>::max();
+        return 0;
+    }
+
+    num_fields = sscanf(str, "%lf%c%c", &value, &units[0], &units[1]);
+    if (num_fields == 1) {
+        per_sec = 1;
+    } else if ((num_fields == 2) || (num_fields == 3)) {
+        if (!strcmp(units, "h")) {
+            per_sec = 1.0 / 3600.0;
+        } else if (!strcmp(units, "m")) {
+            per_sec = 1.0 / 60.0;
+        } else if (!strcmp(units, "s")) {
+            per_sec = 1;
+        } else if (!strcmp(units, "ms")) {
+            per_sec = 1e3;
+        } else if (!strcmp(units, "us")) {
+            per_sec = 1e6;
+        } else if (!strcmp(units, "ns")) {
+            per_sec = 1e9;
+        } else {
+            return -1;
+        }
+    } else {
+        return -1;
+    }
```

Comment:
just pass number of seconds.. and i think glibc already has some kind of time parser, see `man ctime`

Reply:
> just pass number of seconds.

btw, most likely we will set 1h or 0.5h instead of a few seconds
probably, better have to parse time units? will replace with glibc's one